docs/adr/0008-managementv3-clusters-support.md

```diff
@@ -0,0 +1,42 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [8. Management V3 clusters support](#8-management-v3-clusters-support)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 8. Management V3 clusters support
+
+* Status: proposed
+* Date: 2024-01-25
+* Authors: @salasberryfin
+* Deciders: @richardcase @alexander-demicev @furkatgofurov7 @Danil-Grigorev @mjura
+
+## Context
+
+As an operator, I'd like to have an alternative import strategy that will generate a `management.cattle.io/v3` cluster instead of a `provisioning.cattle.io/v1` cluster. This is to ensure that we can support this scenario in light of future changes to Rancher.
+
+## Decision
+
+The current import controller logic creates a `provisioning.cattle.io/v1` cluster when a new CAPI cluster is detected. To support a new import strategy, based on a new feature flag, Rancher Turtles will either start the existing `provisioning.cattle.io` controller or a new one for `management.cattle.io`.
+
+Adding support for this resource presents some challenges:
+- Owner Reference cannot be set: `management.cattle.io/v3` clusters are cluster-scoped while CAPI clusters are namespaced.
+- Display name and resource name are different: the resource name must satisfy the regex `^(c-[a-z0-9]{5}|local)$`.
+- Cluster registration token exists in a namespace with the same name as the `management.cattle.io/v3` cluster.
+
+Of all three, the naming strategy is the main concern. To be able to control the status of the cluster during reconciliation, there must be a deterministic name that identifies the resource. The decision is that we can create a hash from the cluster's display name and truncate it to the required five characters. This operation has low cost and a sufficiently randomized result yet it is deterministic.
+
+From an end-user perspective:
+
+* The default behavior remains unchanged. If no feature flag is set, the current `provisioning.cattle.io` import strategy will be used.
+* User provided name will be used in Rancher dashboard while the underlying `management.cattle.io/v3` wil have a name that satisfies Rancher's naming requirement: `c-` followed by 5 semi-random characters.
+
+## Consequences
+
+- Kubernetes' native garbage collector using the owner reference chain is not a viable option due to the namespaced vs global scoped conflict. This means that there needs to be a custom logic to manage deletion of resources.
```

Comment:
Ownership references can't be set from the namespace scoped object, but it can be set from cluster scoped. This proposal has larger implications on managed / non-managed scenario. We still need to set ownerships to CAPI cluster from V3 cluster for managed, and don't set for non-managed.

Reply:
Based on the note on this document https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/, owner/dependent must either be both in the same namespace or both cluster-scoped.

Reply:
The doc is incorrect in that assumption. Only namespaced resource can’t set cross-namespace owner reference, not the cluster scoped.